From: Usama Arif <usamaarif642@gmail.com>
Subject: mm: do not start/end writeback for pages stored in zswap
Date: Mon, 10 Jun 2024 15:30:37 +0100

start/end writeback combination incorrectly increments NR_WRITTEN counter,
eventhough the pages aren't written to disk.  Pages successfully stored in
zswap should just unlock folio and return from writepage.

Link: https://lkml.kernel.org/r/20240610143037.812955-1-usamaarif642@gmail.com
Signed-off-by: Usama Arif <usamaarif642@gmail.com>
Cc: Chengming Zhou <chengming.zhou@linux.dev>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: kernel-team@meta.com
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: Nhat Pham <nphamcs@gmail.com>
Cc: Yosry Ahmed <yosryahmed@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_io.c |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/page_io.c~mm-do-not-start-end-writeback-for-pages-stored-in-zswap
+++ a/mm/page_io.c
@@ -196,9 +196,7 @@ int swap_writepage(struct page *page, st
 		return ret;
 	}
 	if (zswap_store(folio)) {
-		folio_start_writeback(folio);
 		folio_unlock(folio);
-		folio_end_writeback(folio);
 		return 0;
 	}
 	if (!mem_cgroup_zswap_writeback_enabled(folio_memcg(folio))) {
_
